src/ModelObjects/component.rs

```diff
@@ -382,11 +374,11 @@ impl Component {
                     constraint_applyer::apply_constraints_to_state2(
                         target_inv,
                         &mut new_state,
-                        currState.get_dimensions(),
+                        currState.zone.dimension,
```

Comment:
looking at constraint_applyer::apply_constraints_to_state2 it doesn't ever use the dimension, we should remove it from its arguments so we dont have to do stuff like this

Reply:
This has been fixed